src/components/NavBar.js

```diff
@@ -36,7 +36,7 @@ const NavBar = () => {
   // set the current user by finding it in the user list based on the current identity
   let currentUser = user;
   if (identity && users) {
-    const u = users.find(u => u.id === identity);
+    const u = users.find(u => u.key === identity || u.properties?.email === identity);
```

Comment:
Does the line above ensure that we can find the user by email even before the user's key has changed from the ugly Dex identifier to the email?

Reply:
Yes, it will first try to match the identity(email) with the `user.key` which currently is the dex identifier, the match will be false but it will match with `u.properties?.email`